core/src/banking_stage/forwarder.rs

```diff
@@ -22,28 +22,46 @@ use {
     },
 };
 
-pub struct Forwarder;
+pub(crate) struct Forwarder {
+    poh_recorder: Arc<RwLock<PohRecorder>>,
+    bank_forks: Arc<RwLock<BankForks>>,
+    socket: UdpSocket,
```

Comment:
It is not clear what this socket is used for -- can you add a comment here?

Reply:
Didn't change how this works, just made the `Forwarder` store own it instead of the BankingStage thread. That said it appears to be used here for Forwarding Tpu Votes:
```rust
            let res = if let ForwardOption::ForwardTpuVote = forward_option {
                // The vote must be forwarded using only UDP.
                banking_stage_stats
                    .forwarded_vote_count
                    .fetch_add(packet_vec_len, Ordering::Relaxed);
                let pkts: Vec<_> = packet_vec.into_iter().zip(repeat(addr)).collect();
                batch_send(socket, &pkts).map_err(|err| err.into())
            } else {
                // All other transactions can be forwarded using QUIC, get_connection() will use
                // system wide setting to pick the correct connection object.
                banking_stage_stats
                    .forwarded_transaction_count
                    .fetch_add(packet_vec_len, Ordering::Relaxed);
                let conn = connection_cache.get_connection(&addr);
                conn.send_data_batch_async(packet_vec)
            };
```